src/vsr/journal.zig

```diff
@@ -1415,28 +1460,35 @@ pub fn Journal(comptime Replica: type, comptime Storage: type) type {
                     // TODO Explore scenarios where the data on disk may resurface after a crash.
                     const slot = self.slot_for_op(header.op);
                     assert(slot.index == index);
-                    self.headers[slot.index] = Header.reserved(replica.cluster, slot.index);
-                    self.dirty.clear(slot);
-                    self.faulty.clear(slot);
-                    // Do not clear `prepare_inhabited`/`prepare_checksums`. The prepare is
-                    // untouched on disk, and may be useful later. Consider this scenario:
-                    //
-                    // 1. Op 4 is received; start writing it.
-                    // 2. Op 4's prepare is written (setting `prepare_checksums`), start writing
-                    //    the headers.
-                    // 3. View change. Op 4 is discarded by `remove_entries_from`.
-                    // 4. View change. Op 4 (the same one from before) is back, marked as dirty. But
-                    //    we don't start a write, because `journal.writing()` says it is already in
-                    //    progress.
-                    // 5. Op 4's header write finishes (`write_prepare_on_write_header`).
-                    //
-                    // If `remove_entries_from` cleared `prepare_checksums`,
-                    // `write_prepare_on_write_header` would clear `dirty`/`faulty` for a slot with
-                    // `prepare_inhabited=false`.
+                    self.remove_entry(slot);
                 }
             }
         }
 
+        pub fn remove_entry(self: *Self, slot: Slot) void {
+            const replica = @fieldParentPtr(Replica, "journal", self);
+
+            self.headers[slot.index] = Header.reserved(replica.cluster, slot.index);
+            self.headers_redundant[slot.index] = self.headers[slot.index];
```

Comment:
We could make these lines safer w.r.t. line ordering by caching `reserved = Header.reserved` and then using it for both.